services/audio/src/main/java/ca/bc/gov/tno/services/audio/handlers/FetchDataService.java

```diff
@@ -0,0 +1,202 @@
+    this.eventPublisher = eventPublisher;
+  }
+
+  /**
+   * Start an audio stream, if not already in progress, and extract an audio clip. Publish an audio event so that the Kafka 
+   * Producer can push the content to a topic.
+   */
+  @Override
+  @EventListener
+  public void onApplicationEvent(TransactionBeginEvent<AudioConfig> event) {
+
+    audioConfig = event.getConfig();
+    String mediaSource = audioConfig.getCaptureDir() + "/" + audioConfig.getId() + ".mpg";
+    String cmd = audioConfig.getClipCmd();
+    String destination = clipPath();
+
+    try {
+      caller = event.getSource();
+      var url = audioConfig.getAudioUrl();
+      var readyEvent = new ProducerSendEvent(caller, audioConfig, destination);
+
+      if (startAudioStream()) {
+        eventPublisher.publishEvent(readyEvent);
+      } else {
+        executeClipCmd(cmd, mediaSource, destination);
+        eventPublisher.publishEvent(readyEvent);
+      }
+    } catch (Exception ex) {
+      var errorEvent = new ErrorEvent(this, ex);
+      eventPublisher.publishEvent(errorEvent);
+    }
+  }
+
+  /**
+   * Obtain the location of the streaming file for this event and check that it is being written to.
+   * If the file doesn't exist, or the audio stream has dropped, start streaming from the media source.
+   */
+  private boolean startAudioStream() {
+
+    String captureFilePath = audioConfig.getCaptureDir() + "/" + audioConfig.getId() + ".mpg";
+    File captureFile = new File(captureFilePath);
+    String cmd = audioConfig.getCaptureCmd();
+    long created = 0;
+    boolean startedStream = false;
+
+    // Start recording the audio stream if it doesn't exist
+    if (!captureFile.exists()) {
+        executeCaptureCmd(cmd, captureFilePath);
+        audioConfig.setStreamStartTime(System.currentTimeMillis());
+        startedStream = true;
+    } else {
+      // Check to see if the audio stream has dropped
+      // TODO: handle rollover from one day to next
+      long modified = captureFile.lastModified();
+      long current = System.currentTimeMillis();
+      if (current - modified > 60000) {
+        executeCaptureCmd(cmd, captureFilePath);
+        audioConfig.setStreamStartTime(System.currentTimeMillis());
+        startedStream = true;
+      } 
+    }
+
+    return startedStream;
+  }
+
+  /**
+   * Executes an audio stream capture command as defined in the audioConfig object
+   * 
+   * @param cmd The Linux command to execute 
+   * @param captureFilePath The full path of the output directory
+   */
+  private void executeCaptureCmd(String cmd, String captureFilePath) {
+
+    if (!cmd.equals("")) {
+      cmd = cmd.replace("[capture-url]", audioConfig.getAudioUrl());
+      cmd = cmd.replace("[capture-path]", captureFilePath);
+
+      String[] cmda = {
+        "/bin/sh",
+            "-c",
+        cmd
+      };
+
+      try {
+        Process p = new ProcessBuilder(cmda).start();
```

Comment:
Does running this process block the service thread?  Will a failure in the process after it has started bubble up into this try+catch"?

Reply:
The process variable p is returned without blocking. It can be used to observe the process and see why it has terminated. The existing system does nothing with these variables. A failure in the process can return an exit code but is unlikely to throw an exception.